src/IssueViz.Security/Taint/TaintIssuesBindingMonitor.cs

```diff
@@ -0,0 +1,76 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2020 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.ComponentModel.Composition;
+using System.Threading.Tasks;
+using SonarLint.VisualStudio.Core.Binding;
+
+namespace SonarLint.VisualStudio.IssueVisualization.Security.Taint
+{
+    internal interface ITaintIssuesBindingMonitor : IDisposable
+    {
+        void ListenToBindingChanges();
+    }
+
+    [Export(typeof(ITaintIssuesBindingMonitor))]
+    internal sealed class TaintIssuesBindingMonitor : ITaintIssuesBindingMonitor
+    {
+        private readonly IActiveSolutionBoundTracker activeSolutionBoundTracker;
+        private readonly ITaintIssuesSynchronizer taintIssuesSynchronizer;
+
+        [ImportingConstructor]
+        public TaintIssuesBindingMonitor(IActiveSolutionBoundTracker activeSolutionBoundTracker, ITaintIssuesSynchronizer taintIssuesSynchronizer)
+        {
+            this.activeSolutionBoundTracker = activeSolutionBoundTracker;
+            this.taintIssuesSynchronizer = taintIssuesSynchronizer;
+        }
+
+        public void ListenToBindingChanges()
```

Comment:
I'm not clear on what the lifecycle of this object is;
* who is going to create it and when, and
* when will the `ListenToBindingChanges` method will be called, and will it be called more than once?

The un-registration and re-registration suggests the method is expected to be called more than once, but I don't understand why that would be necessary (also, I thought re-registering the same event handler callback multiple times was a no-op, so I'm not sure the un-registration is necessary).

If it's only going to be called once on construction, could it be done in the constructor?